### PR TITLE
test: add focused coverage for attestations and signerverifier options

### PR DESCRIPTION
**Pull request by Demiserular**
## Description

This PR improves test coverage across the `attestations` and `signerverifier` packages by adding targeted unit tests for certificate loading, predicate conversion, and sigstore options. It also extends GPG tests to include better coverage for negative paths and accessors, ensuring more robust error handling and verification across the signer/verifier components.
## AI Usage

- [x] I **did not** use generative AI at all in making the content of this pull
  request.
- [ ] I **did** use generative AI in some form in making the content of this
  pull request. I have described my use of AI below.

## Contributor Checklist

- [x] I **have manually reviewed all content** submitted to gittuf in this pull
  request.
- [x] I fully understand the content I am submitting.
- [x] The changes introduced are documented and have tests included if
  applicable.
- [x] My changes do not infringe on copyright/trademarks/etc.
- [x] All commits in this pull request include a [DCO
  Signoff](https://wiki.linuxfoundation.org/dco).
- [x] By submitting this pull request, I agree to follow the gittuf [Code of
  Conduct](https://github.com/gittuf/community/blob/main/CODE-OF-CONDUCT.md).

### CODE REVIEW

**Comment by Demiserular** [comment]
Hi @patzielinski, please review this tests-only PR when you have a moment.
It adds coverage for `common`, `sigstore/options/signer`, `sigstore/options/verifier` and also extends GPG negative-path tests.

File: internal/attestations/common/common_test.go
Comment (patzielinski) [inline internal/attestations/common/common_test.go:1]: Tests for this will be added in #1293, please remove.

File: internal/signerverifier/sigstore/options/signer/signer_test.go
Comment (patzielinski) [inline internal/signerverifier/sigstore/options/signer/signer_test.go:1]: We exclude options files from testing, please remove.

File: internal/signerverifier/sigstore/options/verifier/verifier_test.go
Comment (patzielinski) [inline internal/signerverifier/sigstore/options/verifier/verifier_test.go:1]: We exclude options files from testing, please remove.

File: internal/signerverifier/common/common_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoadCertsFromPath(t *testing.T) {
+	t.Run("file not found", func(t *testing.T) {
+		_, err := LoadCertsFromPath(filepath.Join(t.TempDir(), "missing.pem"))
+		require.Error(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]: Use `assert` for cases that you are testing. Please also use `assert.ErrorIs` if the error is defined in the codebase (e.g. `Err...`) or `assert.ErrorContains` if you can match on part of the error that is constant.

Please apply these changes throughout the entire PR.

File: internal/signerverifier/common/common_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	})
+
+	t.Run("invalid pem", func(t *testing.T) {
+		pemPath := filepath.Join(t.TempDir(), "invalid.pem")
+		require.NoError(t, os.WriteFile(pemPath, []byte("not a cert"), 0o600))
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]: Don't inline the `WriteFile` operation; assign an error var and then use `require.Nil` on the var.

File: internal/signerverifier/common/common_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		require.Error(t, err)
+	})
+
+	t.Run("success", func(t *testing.T) {
+		pemPath := filepath.Join(t.TempDir(), "cert.pem")
+		require.NoError(t, os.WriteFile(pemPath, createTestCertificatePEM(t), 0o600))
+
+		certs, err := LoadCertsFromPath(pemPath)
+		require.NoError(t, err)
+		require.Len(t, certs, 1)
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]: Can this be a stronger assert, e.g. matching the contents or some field that is expected to match some value?

**Review by Demiserular — COMMENTED** [review]
I was on windows 🥲

File: internal/signerverifier/common/common_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		NotAfter:              time.Now().Add(1 * time.Hour),
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		BasicConstraintsValid: true,
+	}
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, tmpl, tmpl, privateKey.Public(), privateKey)
+	require.NoError(t, err)
+
+	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: derBytes})
+}
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go:70]: Can this be replaced with using a fixed PEM key already in (or add a new one to) `internal/testartifacts`?

File: internal/signerverifier/gpg/gpg_test.go [hunk 5]
```diff
@@ -82,3 +84,54 @@ func TestLoadGPGKeyFromBytes(t *testing.T) {
 	assert.Equal(t, KeyType, key.Scheme)
 	assert.Equal(t, "157507bbe151e378ce8126c1dcfe043cdd2db96e", key.KeyID)
 }
+
+func TestLoadGPGKeyFromBytesInvalid(t *testing.T) {
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]: Please inline into the `TestLoadGPGKeyFromBytes` test.

File: internal/signerverifier/gpg/gpg_test.go [hunk 6]
```diff
@@ -82,3 +84,54 @@ func TestLoadGPGKeyFromBytes(t *testing.T) {
...
+	_, err := LoadGPGKeyFromBytes([]byte("not a gpg key"))
+	require.Error(t, err)
+}
+
+func TestNewVerifierFromKeyAccessors(t *testing.T) {
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
func TestNewVerifierFromKey(t *testing.T) {
```

File: internal/signerverifier/gpg/gpg_test.go [hunk 7]
```diff
@@ -82,3 +84,54 @@ func TestLoadGPGKeyFromBytes(t *testing.T) {
...
+	pubKey, err := LoadGPGKeyFromBytes(artifacts.GPGKey1Public)
+	require.NoError(t, err)
+
+	verifier, err := NewVerifierFromKey(pubKey)
+	require.NoError(t, err)
+
+	keyID, err := verifier.KeyID()
+	require.NoError(t, err)
+	assert.Equal(t, pubKey.KeyID, keyID)
+	assert.NotNil(t, verifier.Public())
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]: Can this be a stronger `assert`? e.g. assert that the public ID matches an expected value?

File: internal/signerverifier/gpg/gpg_test.go [hunk 8]
```diff
@@ -82,3 +84,54 @@ func TestLoadGPGKeyFromBytes(t *testing.T) {
...
+	assert.Equal(t, pubKey, verifier.MetadataKey())
+}
+
+func TestNewVerifierFromKeyInvalid(t *testing.T) {
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]: Please add as a `t.Run` to `TestNewVerifierFromKey` (and make the existing case above another `t.Run`).

File: internal/signerverifier/gpg/gpg_test.go [hunk 9]
```diff
@@ -82,3 +84,54 @@ func TestLoadGPGKeyFromBytes(t *testing.T) {
...
+
+func TestNewSignerFromKeyIDWithInvalidProgram(t *testing.T) {
+	_, err := NewSignerFromKeyID("missing-key", WithGPGProgram("gpg-program-that-does-not-exist"))
+	require.Error(t, err)
+}
+
+func TestGetPublicKeyForKeyIDWithInvalidProgram(t *testing.T) {
+	_, err := getPublicKeyForKeyID("missing-key", "gpg-program-that-does-not-exist")
+	require.Error(t, err)
+}
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]: I'd just remove these for now, these failures would likely be better tested similar to how we wrote tests in the `ssh` package.

**Review by patzielinski — CHANGES_REQUESTED** [review]
Please also update the PR to use the official template, thanks!

**Comment by Demiserular** [comment]
kindly review now @patzielinski

File: internal/signerverifier/common/common_test.go (hunk 4 above)
Comment (patzielinski) [reply internal/signerverifier/common/common_test.go:70]: Deferring this for later.

File: internal/signerverifier/common/common_test.go [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+		assert.ErrorContains(t, err, "PEM decoding")
+	})
+
+	t.Run("success", func(t *testing.T) {
+		pemPath := filepath.Join(t.TempDir(), "cert.pem")
+		err := os.WriteFile(pemPath, createTestCertificatePEM(t), 0o600)
+		require.Nil(t, err)
+
+		certs, err := LoadCertsFromPath(pemPath)
+		assert.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]:
```suggestion
		assert.Nil(t, err)
```

File: internal/signerverifier/common/common_test.go [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+		assert.Len(t, certs, 1)
+	})
+}
+
+func createTestCertificatePEM(t *testing.T) []byte {
+	t.Helper()
+
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]:
```suggestion
	require.Nil(t, err)
```

File: internal/signerverifier/common/common_test.go [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+			CommonName: "gittuf-test-cert",
+		},
+		NotBefore:             time.Now().Add(-1 * time.Hour),
+		NotAfter:              time.Now().Add(1 * time.Hour),
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		BasicConstraintsValid: true,
+	}
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, tmpl, tmpl, privateKey.Public(), privateKey)
+	require.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]:
```suggestion
	require.Nil(t, err)
```

File: internal/signerverifier/gpg/gpg_test.go [hunk 13]
```diff
@@ -74,11 +76,47 @@ func TestGPG(t *testing.T) {
 }
 
 func TestLoadGPGKeyFromBytes(t *testing.T) {
-	keyBytes := artifacts.GPGKey1Public
+	t.Run("valid", func(t *testing.T) {
+		key, err := LoadGPGKeyFromBytes(artifacts.GPGKey1Public)
+		assert.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
		assert.Nil(t, err)
```

File: internal/signerverifier/gpg/gpg_test.go [hunk 14]
```diff
@@ -74,11 +76,47 @@ func TestGPG(t *testing.T) {
...
+	t.Run("valid", func(t *testing.T) {
+		pubKey, err := LoadGPGKeyFromBytes(artifacts.GPGKey1Public)
+		require.NoError(t, err)
+
+		verifier, err := NewVerifierFromKey(pubKey)
+		assert.NoError(t, err)
+		require.NotNil(t, verifier)
+
+		keyID, err := verifier.KeyID()
+		assert.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
		require.Nil(t, err)

		verifier, err := NewVerifierFromKey(pubKey)
		assert.Nil(t, err)
		require.NotNil(t, verifier)

		keyID, err := verifier.KeyID()
		assert.Nil(t, err)
```

**Review by patzielinski — COMMENTED** [review]
Some nits and this should be good to go. Please squash commits down to one when complete, thanks!

**Comment by Demiserular** [comment]
@patzielinski i re-run the jobs here.

**Comment by patzielinski** [comment]
My comments on using `assert.Nil` and `require.Nil` aren't addressed, please address them.

**Comment by patzielinski** [comment]
Just a clarification, I meant that you should use the `.Nil` variants instead of the `NoError` ones. Apologies for the confusion.

**Comment by Demiserular** [comment]
You are the captian of the ship , no apologies required . Its just about few more clicks.

File: internal/signerverifier/common/common_test.go [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
```
Comment (patzielinski) [inline internal/signerverifier/common/common_test.go]:
```suggestion
	require.Nil(t, err)
```

File: internal/signerverifier/gpg/gpg_test.go (hunk 13 above)
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
		assert.Nil(t, err)
```

File: internal/signerverifier/gpg/gpg_test.go [hunk 16]
```diff
@@ -74,11 +76,47 @@ func TestGPG(t *testing.T) {
...
+	t.Run("invalid", func(t *testing.T) {
+		_, err := LoadGPGKeyFromBytes([]byte("not a gpg key"))
+		assert.ErrorContains(t, err, "no armored data found")
+	})
+}
+
+func TestNewVerifierFromKey(t *testing.T) {
+	t.Run("valid", func(t *testing.T) {
+		pubKey, err := LoadGPGKeyFromBytes(artifacts.GPGKey1Public)
+		require.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
		require.Nil(t, err)
```

File: internal/signerverifier/gpg/gpg_test.go [hunk 17]
```diff
@@ -74,11 +76,47 @@ func TestGPG(t *testing.T) {
...
+
+		verifier, err := NewVerifierFromKey(pubKey)
+		assert.NoError(t, err)
```
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
		assert.Nil(t, err)
```

File: internal/signerverifier/gpg/gpg_test.go (hunk 14 above)
Comment (patzielinski) [inline internal/signerverifier/gpg/gpg_test.go]:
```suggestion
		assert.Nil(t, err)
```

**Review by patzielinski — COMMENTED** [review]
Just a few nits. Please squash commits when done and we'll get this merged, thanks!

**Review by patzielinski — APPROVED** [review]
